src/WebUI/Workloads/WorkloadsStore.ts

```diff
@@ -80,24 +84,40 @@ export class WorkloadsStore extends StoreBase {
     }
 
     private _setReplicaSetsList = (replicaSetList: V1ReplicaSetList): void => {
-        this._state.replicaSetList = replicaSetList;
-        this.emit(WorkloadsEvents.ReplicaSetsFetchedEvent, this);
+        const _stateItems = this._state.replicaSetList ? this._state.replicaSetList.items : [];
+        const _replicaItems = replicaSetList ? replicaSetList.items : [];
+        const isEquals = Utils.isDeepEquals(_stateItems, _replicaItems);
+        if (!isEquals || !this._state.replicaSetList) {
+            this._state.replicaSetList = replicaSetList;
+            this.emit(WorkloadsEvents.ReplicaSetsFetchedEvent, this);
+        }
         if (this._state.replicaSetList && this._state.replicaSetList.items && this._state.replicaSetList.items.length > 0) {
```

Comment:
should not we keep this inside the if condition of "if (!isEquals || !this._state.replicaSetList) {"?